compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -1708,7 +1708,11 @@ impl<'tcx> InferCtxtPrivExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
                 ty::Foreign(..) => Some(18),
                 ty::GeneratorWitness(..) => Some(19),
                 ty::GeneratorWitnessMIR(..) => Some(20),
-                ty::Placeholder(..) | ty::Bound(..) | ty::Infer(..) | ty::Error(_) => None,
+                ty::Alias(ty::Weak, ..)
```

Comment:
I guess it doesn't hurt to give them their own type category..... idk, either way works  but it's inconsistent I feel. Projections, Inherents, and Weaks should all be handled the same way.